subsystems/DrivetrainBase.java

```diff
@@ -1,5 +1,65 @@
 package org.frc5587.lib.subsystems;
 
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+
 public class DrivetrainBase {
```

Comment:
This should be an abstract class and extend PIDSubsystem. Keep in mind that you'll have to implemented the required commands for that class, along with the position and velocity methods we use. I would look at the Drivetrain class from GameChangers for an idea of what methods you need. Also, look at FixedHoodedShooter within JavaLibs for an idea of the structure you want to use within abstract classes.